### PR TITLE
Add upgrade guide for v8

### LINKED ISSUE #3516: Create Upgrade Guide for v8 release

### PR DESCRIPTION
**Pull request by kmaehashi**
Closes #3516

This guide is generated from no-compat labelled PRs: https://github.com/cupy/cupy/issues?q=label%3Ano-compat

Note that no-compat changes between v8 pre-releases (alpha/beta/rc) are excluded from this guide.

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docs/source/upgrade.rst [hunk 1]
```diff
@@ -5,21 +5,51 @@ Upgrade Guide
...
 
-CuPy official Docker images (see :doc:`install` for details) are now updated to use CUDA 10.2 and Python 3.6.
+* CuPy official Docker images (see :doc:`install` for details) are now updated to use CUDA 10.2 and Python 3.6.
+* SciPy and Optuna are now pre-installed.
+
+CUB Support and Compiler Requirement
+------------------------------------
+
+CUB module is now built by default.
+You can enable the use of CUB by setting ``CUPY_ACCERELATORS="cub"`` (see :doc:`reference/environment` for details).
```
Comment (leofang) [inline docs/source/upgrade.rst:33]:
```suggestion
You can enable the use of CUB by setting ``CUPY_ACCELERATORS="cub"`` (see :doc:`reference/environment` for details).
```

File: docs/source/upgrade.rst [hunk 2]
```diff
@@ -5,21 +5,51 @@ Upgrade Guide
...
+
+Due to this change, g++-6 or later is required when building CuPy from the source.
+See :doc:`install` for details.
+
+The following environment variables are no longer effective:
+
+* ``CUB_DISABLED``: Use ``CUPY_ACCERELATORS`` as aforementioned.
```
Comment (leofang) [inline docs/source/upgrade.rst:40]:
```suggestion
* ``CUB_DISABLED``: Use ``CUPY_ACCELERATORS `` as aforementioned.
```

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docs/source/upgrade.rst [hunk 3]
```diff
@@ -5,21 +5,51 @@ Upgrade Guide
...
+CUB module is now built by default.
+You can enable the use of CUB by setting ``CUPY_ACCELERATORS="cub"`` (see :doc:`reference/environment` for details).
+
+Due to this change, g++-6 or later is required when building CuPy from the source.
+See :doc:`install` for details.
+
+The following environment variables are no longer effective:
+
+* ``CUB_DISABLED``: Use ``CUPY_ACCELERATORS`` as aforementioned.
+* ``CUB_PATH``: No longer required as CuPy uses CUB source tree bundled with CUDA (only when using CUDA 11.0 or later) or CuPy distribution.
```
Comment (emcastillo) [inline docs/source/upgrade.rst:41]:
```suggestion
* ``CUB_PATH``: No longer required as CuPy uses either the CUB source bundled with CUDA (only when using CUDA 11.0 or later) or the one in the CuPy distribution.
```
I am fine with leaving the previous one, this is just style so I can't say which is better, pick up the one you like most 😁

**Review by emcastillo — APPROVED** [review]

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**